src/fparser/two/tests/test_c99preprocessor.py

```diff
@@ -0,0 +1,524 @@
+#!/usr/bin/env python
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test C99 Preprocessor directives: This file tests the preprocessor
+directives defined in the C99 standard. Whilst this is not part of any
+Fortran standard, most compilers support preprocessing of Fortran files
+and support the majority of directives from the C-Standard. Since this
+is actively used in user codes there are cases where users might like to
+keep the directives in the Fortran parse tree and output it again.
+
+'''
+# Author: Balthasar Reuter <balthasar.reuter@ecmwf.int>
+# Based on previous work by Martin Schlipf (https://github.com/martin-schlipf)
+# First version created: Jan 2020
+
+import pytest
+from fparser.two.C99Preprocessor import (Cpp_If_Stmt, Cpp_Elif_Stmt,
+                                         Cpp_Else_Stmt, Cpp_Endif_Stmt,
+                                         Cpp_Include_Stmt, Cpp_Macro_Stmt,
+                                         Cpp_Macro_Identifier, Cpp_Undef_Stmt,
+                                         Cpp_Line_Stmt, Cpp_Error_Stmt,
+                                         Cpp_Warning_Stmt, Cpp_Null_Stmt)
+from fparser.two.parser import ParserFactory
+from fparser.two.utils import NoMatchError
+from fparser.api import get_reader
+
+
+def test_if_stmt(f2003_create):
```

Comment:
Use of f2003_create causes pylint to complain so we now suggest using the following instead:
`@pytest.mark.usefixtures("f2003_create")`